### PR TITLE
support async loop (branches and lanes) by slang events

### LINKED ISSUE #352: missing appropriate events for async-loop for supporting drill down

### PR DESCRIPTION
**Pull request by shajyhia**
Fixes #352

### CODE REVIEW

**Comment by oritstone** [comment]
@Bonczidai please review

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java [hunk 1]
```diff
@@ -17,17 +17,19 @@
...
+import io.cloudslang.lang.runtime.env.Context;
+import io.cloudslang.lang.runtime.env.ReturnValues;
+import io.cloudslang.lang.runtime.env.RunEnvironment;
 import io.cloudslang.lang.runtime.events.LanguageEventData;
-import org.apache.commons.lang.SerializationUtils;
-import org.apache.log4j.Logger;
 import io.cloudslang.score.api.EndBranchDataContainer;
 import io.cloudslang.score.api.execution.ExecutionParametersConsts;
 import io.cloudslang.score.lang.ExecutionRuntimeServices;
+import org.apache.commons.lang.SerializationUtils;
```
Comment (Bonczidai) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java]: maybe you could import `SerializationUtils` from `org.apache.commons.lang3` just to use the newer version. WDYT?

Comment (shajyhia) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java]: if it continue to work then be my guest

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java [hunk 2]
```diff
@@ -90,11 +101,10 @@ public void addBranches(@Param(ScoreLangConstants.ASYNC_LOOP_STATEMENT_KEY) Asyn
                         nextStepId,
                         branchBeginStep);
 
-                fireEvent(executionRuntimeServices, ScoreLangConstants.EVENT_BRANCH_START,
-                        "async loop branch created", runEnv.getExecutionPath().getCurrentPath(),
-                        LanguageEventData.StepType.TASK, nodeName, Pair.of(ScoreLangConstants.REF_ID, refId),
-                        Pair.of(RuntimeConstants.SPLIT_ITEM_KEY, splitItem));
+                // take path down up level
+                runEnv.getExecutionPath().up();
```
Comment (Bonczidai) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java]: typo here? should be  `// take path up one level`

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java [hunk 3]
```diff
@@ -226,7 +236,10 @@ private void collectBranchesData(
             ReturnValues executableReturnValues = branchRuntimeEnvironment.removeReturnValues();
             branchesResult.add(executableReturnValues.getResult());
 
-            fireEvent(executionRuntimeServices, runEnv, ScoreLangConstants.EVENT_BRANCH_END,
+            // down branch path
+            branchRuntimeEnvironment.getExecutionPath().up();
```
Comment (Bonczidai) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java]: `// up branch path` right?

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java (hunk 1 above)
Comment (Bonczidai) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java]: let's leave it like this for now

**Comment by Bonczidai** [comment]
:+1: 
@shajyhia please fix the two comments regarding descriptions so we can merge this PR

**Comment by shajyhia** [comment]
done

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java (hunk 3 above)
Comment (shajyhia) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java]: done

**Comment by Bonczidai** [comment]
great, thanks

**Pull request merged**